src/shared_modules/indexer_connector/tests/component/indexerConnector_test.cpp

```diff
@@ -0,0 +1,567 @@
+                                          ++callbackCalled;
+
+                                          if (1 == callbackCalled)
+                                          {
+                                              templateData = std::move(responseData);
+                                              return;
+                                          }
+
+                                          indexData = std::move(responseData);
+                                      }};
+    m_indexerServers[A_IDX]->setInitTemplateCallback(checkInitDataCallback);
+    m_indexerServers[A_IDX]->setInitIndexCallback(checkInitDataCallback);
+
+    // Create connector and wait until the connection is established.
+    nlohmann::json indexerConfig;
+    indexerConfig["name"] = INDEXER_NAME;
+    indexerConfig["hosts"] = nlohmann::json::array({A_ADDRESS});
+    auto indexerConnector {IndexerConnector(indexerConfig, TEMPLATE_FILE_PATH, logFunction, INDEXER_TIMEOUT)};
+    ASSERT_NO_THROW(waitUntil([this]() { return m_indexerServers[A_IDX]->initialized(); }, MAX_INDEXER_INIT_TIME_MS));
+
+    // Check init data.
+    constexpr auto EXPECTED_CALLBACK_CALLED_TIMES {2};
+    ASSERT_EQ(callbackCalled, EXPECTED_CALLBACK_CALLED_TIMES);
+    ASSERT_EQ(templateData, TEMPLATE_DATA);
+    ASSERT_EQ(indexData, TEMPLATE_DATA.at("template"));
+}
+
+/**
+ * @brief Test the connection to an available server with user and password data.
+ *
+ * @note The credentials are dummy ones and there are no functionality checks here. The target of this test is to
+ * increase the test coverage.
+ *
+ */
+TEST_F(IndexerConnectorTest, ConnectionWithUserAndPassword)
+{
+    nlohmann::json indexerConfig;
+    indexerConfig["name"] = INDEXER_NAME;
+    indexerConfig["hosts"] = nlohmann::json::array({A_ADDRESS});
+    indexerConfig["username"] = "user";
+    indexerConfig["password"] = "password";
+
+    // Create connector and wait until the connection is established.
+    auto indexerConnector {IndexerConnector(indexerConfig, TEMPLATE_FILE_PATH, logFunction, INDEXER_TIMEOUT)};
+    EXPECT_NO_THROW(waitUntil([this]() { return m_indexerServers[A_IDX]->initialized(); }, MAX_INDEXER_INIT_TIME_MS));
+}
+
+/**
+ * @brief Test the connection to an available server with SSL credentials.
+ *
+ * @note The SSL data is a dummy one and there are no functionality checks here. The target of this test is to increase
+ * the test coverage.
+ *
+ */
+TEST_F(IndexerConnectorTest, ConnectionWithSslCredentials)
+{
+    nlohmann::json indexerConfig;
+    indexerConfig["name"] = INDEXER_NAME;
+    indexerConfig["hosts"] = nlohmann::json::array({A_ADDRESS});
+    indexerConfig["ssl"]["certificate_authorities"] = nlohmann::json::array({"/etc/filebeat/certs/root-ca.pem"});
+    indexerConfig["ssl"]["certificate"] = "/etc/filebeat/certs/filebeat.pem";
+    indexerConfig["ssl"]["key"] = "/etc/filebeat/certs/filebeat-key.pem";
```

Comment:
Question: Are these files read/write? If so, couldn't this be a problem?

Reply:
These are dummy file paths. The tests are performed over HTTP (not HTTPS) so they are not really used. In fact, they may not exist and the tests will pass.

The addition of these configurations was just to increase the test coverage of the class.

Reply:
Got it! Maybe we could add a comment just for future testers

Reply:
This [comment](https://github.com/wazuh/wazuh/blob/7658be87afbf8650dae86b774ffd16c3e6929c60/src/shared_modules/indexer_connector/tests/component/indexerConnector_test.cpp#L208) is already present.

Do you think it could be improved?